packages/client/lib/service/fullethereumservice.ts

```diff
@@ -94,6 +95,11 @@ export class FullEthereumService extends EthereumService {
       peer.eth!.send('BlockBodies', { reqId, bodies })
     } else if (message.name === 'NewBlockHashes') {
       this.synchronizer.handleNewBlockHashes(message.data)
+    } else if (message.name === 'NewBlock') {
+      const common = this.config.chainCommon.copy()
+      common.setHardforkByBlockNumber(this.chain.blocks.height)
+      const block = Block.fromValuesArray(message.data[0], { common })
+      await this.synchronizer.handleNewBlock(block)
```

Comment:
Ah, two things here.

1. This is double decoding. The decoding and block creation is already done in the `decode()` method from `ethprotocol.ts` - you can test this by do a `console.log()` output on the `message` object. So you are actually creating the block twice. So the block creation from here should be removed and limited to the `decode()` function.
2. Setting the HF with `this.chain.blocks.height` doesn't do the trick respectively leads to wrong results, since our own chain might be behind the tip of the chain. Then you will set the wrong HF. Therefore the suggestion to use the `syncTargetHeight`. On a second I realized though that this is a) - while being a bit better, lol - also not hitting it precisely since there can be a HF switch after `syncTargetHeight` which also isn't caught with this but luckily also b) that I didn't see the forest for all the trees and now remembered that you can just use the `hardforkByBlockNumber` option which should be exactly what you want. 😛

Reply:
Ah, I see this now though odd that it wasn't throwing before.  Using just the  `hardforkByBlockNumber` parameter without providing a common instance in the `decode` function doesn't seem to work (at least on my private network where I'm way passed the London hardfork and it's throwing the error about base fee only can be set when 1559 is activated).  I need to pass a `common` instance in that aligns with the current state of the chain and I don't think the `common` that's available by default in the `ethProtocol` is the same as the chain.